crates/turbopack/src/condition.rs

```diff
@@ -27,12 +30,31 @@ impl ContextCondition {
         ContextCondition::Not(Box::new(condition))
     }
 
+    #[async_recursion]
     /// Returns true if the condition matches the context.
-    pub fn matches(&self, context: &FileSystemPath) -> bool {
+    pub async fn matches(&self, context: &FileSystemPath) -> bool {
         match self {
-            ContextCondition::All(conditions) => conditions.iter().all(|c| c.matches(context)),
-            ContextCondition::Any(conditions) => conditions.iter().any(|c| c.matches(context)),
-            ContextCondition::Not(condition) => !condition.matches(context),
+            ContextCondition::All(conditions) => {
+                stream::iter(conditions)
+                    .all(|c| async move { c.matches(context).await })
+                    .await
+            }
+            ContextCondition::Any(conditions) => {
+                stream::iter(conditions)
+                    .any(|c| async move { c.matches(context).await })
+                    .await
+            }
+            ContextCondition::Not(condition) => !condition.matches(context).await,
+            ContextCondition::InPath(path) => {
+                if let Ok(path) = path.await {
+                    path.fs
+                        .root()
+                        .await
+                        .map_or(false, |root| context.is_inside(&root))
```

Comment:
This ignores the error here, but it's better to propagate the error upwards

Reply:
So change `fn matches(...) -> bool` signature to `fn matches(..) -> Result<bool>`?